pkg/controller/function/function_controller.go

```diff
@@ -148,49 +275,244 @@ func (r *ReconcileFunction) Reconcile(request reconcile.Request) (reconcile.Resu
 		data["package.json"] = fn.Spec.Deps
 	}
 
+	return data
+
+}
+
+// Create Function's ConfigMap
+func (r *ReconcileFunction) createFunctionConfigMap(foundCm *corev1.ConfigMap, deployCm *corev1.ConfigMap, fn *runtimev1alpha1.Function) error {
+
+	// Create Function Handler
+	deployCm.Data = createFunctionHandlerMap(fn)
+
 	// Managing a ConfigMap
-	deployCm := &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Labels:    fn.Labels,
-			Namespace: fn.Namespace,
-			Name:      fn.Name,
-		},
-		Data: data,
+	deployCm.ObjectMeta = metav1.ObjectMeta{
+		Labels:    fn.Labels,
+		Namespace: fn.Namespace,
+		Name:      fn.Name,
 	}
+
 	if err := controllerutil.SetControllerReference(fn, deployCm, r.scheme); err != nil {
-		return reconcile.Result{}, err
+		return err
 	}
 
-	foundCm := &corev1.ConfigMap{}
-	err = r.Get(context.TODO(), types.NamespacedName{Name: deployCm.Name, Namespace: deployCm.Namespace}, foundCm)
+	err := r.Get(context.TODO(), types.NamespacedName{Name: deployCm.Name, Namespace: deployCm.Namespace}, foundCm)
 	if err != nil && errors.IsNotFound(err) {
-		log.Info("Creating ConfigMap", "namespace", deployCm.Namespace, "name", deployCm.Name)
+		log.Info("Creating the Function's ConfigMap", "namespace", deployCm.Namespace, "name", deployCm.Name)
 		err = r.Create(context.TODO(), deployCm)
 		if err != nil {
-			return reconcile.Result{}, err
+			return err
 		}
+
+		return nil
 	} else if err != nil {
-		return reconcile.Result{}, err
+		return err
 	}
 
+	log.Info("Function ConfigMap:", "namespace", deployCm.Namespace, "name", deployCm.Name)
+
+	return nil
+}
+
+// Update found Function's ConfigMap
+func (r *ReconcileFunction) updateFunctionConfigMap(foundCm *corev1.ConfigMap, deployCm *corev1.ConfigMap) error {
+
 	if !reflect.DeepEqual(deployCm.Data, foundCm.Data) {
 		foundCm.Data = deployCm.Data
-		log.Info("Updating ConfigMap", "namespace", deployCm.Namespace, "name", deployCm.Name)
-		err = r.Update(context.TODO(), foundCm)
+		// TODO: why is this required ??
```

Comment:
We should remove all added todos unless there is a follow up story for that with a issue number referenced